support/cas-server-support-saml-idp/src/test/java/org/apereo/cas/support/saml/services/SamlIdPServicesManagerRegisteredServiceLocatorTests.java

```diff
@@ -69,4 +74,37 @@ public void verifyReverseOperation() {
         assertNotNull(result);
         assertTrue(result instanceof SamlRegisteredService);
     }
+
+    /**
+     * serviceLocator should not trigger metadata lookups when requested entityID does not match pattern for service in question.
+     *
+     * This test verifies that, in the case of one service entry that does not match the requested entityID, no
+     * metadata lookups are performed.
+     *
+     * @author Hayden Sartoris
+     */
+    @Test
+    public void verifyEntityIDFilter() {
+        SamlRegisteredServiceCachingMetadataResolver resolver = new BrokenMetadataResolver();
+        SamlIdPServicesManagerRegisteredServiceLocator locator = new SamlIdPServicesManagerRegisteredServiceLocator(resolver);
+
+        val service1 = RegisteredServiceTestUtils.getRegisteredService("urn:abc:def.+");
+        service1.setEvaluationOrder(9);
+        val entityID = "https://sp.testshib.org/shibboleth-sp";
+        val service = webApplicationServiceFactory.createService(entityID);
+
+        locator.locate(List.of(service1), service, r -> r.matches(entityID));
+    }
+
+    private static class BrokenMetadataResolver implements SamlRegisteredServiceCachingMetadataResolver {
+        public MetadataResolver resolve(SamlRegisteredService service, CriteriaSet criteriaSet) {
+            throw new IllegalStateException("This method shouldn't have been called");
+        }
+
+        public void invalidate() {
+        }
+
+        public void invalidate(SamlRegisteredService s, CriteriaSet c) {
+        }
```

Comment:
You don't need this. Use mocks, and then do when(myresolver.resolve(...)).thenThrow()...